external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -240,7 +240,9 @@ private boolean commit() {
 
     private boolean poll() {
         final int maxUncommittedOffsets = kafkaSpoutConfig.getMaxUncommittedOffsets();
-        final boolean poll = !waitingToEmit() && numUncommittedOffsets < maxUncommittedOffsets;
+        final boolean poll = !waitingToEmit() &&
+            (numUncommittedOffsets < maxUncommittedOffsets ||
+            !retryService.retriableTopicPartitions().isEmpty());
```

Comment:
`!retryService.retriableTopicPartitions().isEmpty()` can't this condition cause unbounded  polling (i.e. continue forever), if some specific tuples are always failing?

The goal is to have a limit in how much can be polled (hence kept in memory) in case of recurring failures.